### PR TITLE
Enhancement/end game panel new design

### PR DESCRIPTION
**Pull request by wbeltran777**
Almost done with the end game panel, going to finish tomorrow. The panel pops up, score shows, and sprite changes. Currently working on showing the position string for each number.

### CODE REVIEW

**Comment by Gursaanj** [comment]
[Update the end game panel UI and player score](https://trello.com/c/NmWQAqGr/214-update-the-end-game-panel-ui-and-player-score)

File: Assets/Script/UI/EndGamePanel.cs [hunk 1]
```diff
@@ -16,6 +20,49 @@ public class EndGamePanel : BasePanel
...
+    [SerializeField]
+    private GameObject m_carTemplate;
+
+    private List<GameObject> m_listOfCarPanels = new List<GameObject>();
+
+    private List<string> m_possiblePlayerPlacements = new List<string>();
+
+    protected override void Awake()
+    {
+        base.Awake();
```
Comment (m4rtint) [inline Assets/Script/UI/EndGamePanel.cs]: You don't need to override awake if you're just calling the base Awake

File: Assets/Script/UI/EndGamePanel.cs [hunk 2]
```diff
@@ -24,13 +71,18 @@ protected override void Expand()
...
         UpdateScoreText();
     }
 
     private void Start()
     {
         StateManager.OnEndGame += Expand;
+        m_possiblePlayerPlacements.Add("4th");
+        m_possiblePlayerPlacements.Add("3rd");
+        m_possiblePlayerPlacements.Add("2nd");
+        m_possiblePlayerPlacements.Add("1st");
```
Comment (m4rtint) [inline Assets/Script/UI/EndGamePanel.cs]: You can place these strings into the new file 
MainText.cs, which is basically where we store all the UI Text display on screen.

File: Assets/Script/UI/EndGamePanel.cs [hunk 3]
```diff
@@ -43,14 +95,48 @@ private void UpdateEndGameText()
         m_endGameText.text = $"Congratulations {PlayerManager.getPlayerName(ScoreManager.WinningPlayer)} wins!";
```
Comment (m4rtint) [inline Assets/Script/UI/EndGamePanel.cs]: can place this in the MainText.cs too

File: Assets/Script/Managers/ScoreManager.cs [hunk 4]
```diff
@@ -18,6 +18,7 @@ public static class ScoreManager
     private static Dictionary<Player, int> m_carScores;
     private static int m_winningScore = 2;
     private static Player m_winningPlayer;
+    private static string[] m_placementNames = new string[4];
```
Comment (m4rtint) [inline Assets/Script/Managers/ScoreManager.cs]: this is unused

File: Assets/Script/Data/MainText.cs [hunk 5]
```diff
@@ -101,5 +101,44 @@ public static class MainText
         /// </summary>
         public const string PlayerJoinedMessage = "{0} player joined!";
         #endregion
+
+        #region EndGamePanel
+        
+        /// <summary>
+        /// First place string
+        /// </summary>
+        public const string FirstPlace = "1st";
```
Comment (m4rtint) [inline Assets/Script/Data/MainText.cs]: This is fine, but I would personally use this
https://stackoverflow.com/questions/20156/is-there-an-easy-way-to-create-ordinals-in-c

and place in Utility. Just cuz it's cool

**Pull request merged**